source/aws_s3/s3.go

```diff
@@ -21,37 +21,64 @@ func init() {
 
 type s3Driver struct {
 	s3client   s3iface.S3API
-	bucket     string
-	prefix     string
+	config     *Config
 	migrations *source.Migrations
 }
 
+type Config struct {
+	Bucket string
+	Prefix string
+}
+
 func (s *s3Driver) Open(folder string) (source.Driver, error) {
-	u, err := url.Parse(folder)
+	sess, err := session.NewSession()
 	if err != nil {
 		return nil, err
 	}
-	sess, err := session.NewSession()
+
```

Comment:
Parse the URI before creating a new session so malformed URIs won't result in unnecessary work.

Reply:
Done